### ISSUE TITLE
vscode-test@1.0.1 vscodeExecutablePath issue

### ISSUE THREAD

**Issue by Eskibear** [microsoft/vscode-test#31]
To solve issue #29 , you make it point it to VSCode CLI. Now it's ok to install extensions but fail to run tests. 

```ts
// Here the path should point to Electron, otherwise it won't run the test. 
 await runTests({ vscodeExecutablePath, extensionDevelopmentPath, extensionTestsPath });
```

**Comment by octref** [comment]
I published 1.0.2 to temporarily revert the change. I'll work on a new fix.

**Comment by Eskibear** [comment]
may I ask when will the new fix be available?

**Comment by octref** [comment]
Would a change like this work for you?

Before:

```ts
export async function downloadAndUnzipVSCode(version?: DownloadVersion): Promise<string> {}
```

After:
```ts
export async function downloadAndUnzipVSCode(version?: DownloadVersion): Promise<{
	electronPath: string,
	cliPath: string
}> {}
```

**Comment by octref** [comment]
Current issues is:
- Launching process through CLI won't launch tests properly
- Launching process through Electron won't handle args handled in `cli`

So the new API would be:
- For beginners / simple usage, still call `runTests` with the current API. I'll expand `launchArgs`'s documentation to say which args aren't accepted.
- For advanced usage, users call `downloadAndUnzipVSCode` to get the two paths. They still need to pass `vscodeExecutablePath` to `runTests`, but they can run CLI process with `vscodeCLIPath` as well.

**Comment by Eskibear** [comment]
Thanks in advance. The new API can unblock me from the current problem. The request is actually to provide a way to know the CLI path, in order to install prerequisite extensions.

BTW, the [document for vscode-test](https://code.visualstudio.com/api/working-with-extensions/testing-extension#custom-setup-with-vscodetest) is pointing to a dead end, which is harming VS Code's reputation.

**Comment by octref** [comment]
To not break backward-compatibility, I decided to export another function called `resolveCliPathFromExecutablePath`. Sample usage here: https://github.com/microsoft/vscode-test/blob/c7efccb0011a1590fbb1edb819e028a90cb284b6/sample/test/runTest.ts#L71-L75

**Comment by octref** [comment]
Published in 1.1.0.
